src/image/loading_displaying.js

```diff
@@ -138,14 +159,47 @@ p5.prototype.image = function(img, x, y, width, height) {
   //   ]
   // );
 
-  // set defaults
-  x = x || 0;
-  y = y || 0;
-  width = width || img.width;
-  height = height || img.height;
-  var vals = canvas.modeAdjust(x, y, width, height, this._renderer._imageMode);
+  // Validates clipping params. Per drawImage spec sWidth and sHight cannot be
+  // negative or greater than image intrinsic width and height
+  var sAssign = function (sVal, iVal) {
```

Comment:
I think it would be faster to define this function as a helper function inside the module but outside of `p5.prototype.image`. This way it's not redefined every time `image()` is called in each draw loop.